src/conflictresolutionscreen.cpp

```diff
@@ -0,0 +1,138 @@
+/***********************************************************************************
+ *  One Click Installer makes it easy for users to install software, no matter
+ *  where that software is located.
+ *
+ *  Copyright (C) 2016  Shalom <shalomray7@gmail.com>
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ ************************************************************************************
+ *  This program's developed as part of GSoC - 2016
+ *  Project: One Click Installer
+ *  Mentors: Antonio Larrosa, and Cornelius Schumacher
+ *  Organization: OpenSUSE
+ *  Previous Contributor: None
+ ***********************************************************************************/
+
+#include <QApplication>
+#include <QDBusConnection>
+#include <QRadioButton>
+#include "conflictresolutionscreen.h"
+#include "oneclickinstalleradaptor.h"
+
+ConflictResolutionScreen::ConflictResolutionScreen()
+{
+    setStyleSheet( "background-color: white;" );
+    
+    new OneclickinstallerAdaptor( this );
+    QDBusConnection connection = QDBusConnection::systemBus();
+    if ( !connection.isConnected() ) {
+	qFatal( "Cannot connect to the D-Bus system bus" );
+	exit( 1 );
+    }
+    connection.registerObject( "/OCI", this );
+    if ( !connection.registerService("org.opensuse.oneclickinstaller") ) {
+	qDebug() << qPrintable( QDBusConnection::systemBus().lastError().message() );
+        exit( 1 );
+    }
+   
+    m_mainLayout = new QVBoxLayout;
+    
+    //QObject::connect( m_ociHelper, SIGNAL( displayProblemAndSolutions( QString, QString ) ), this, SLOT( problemSolutionWidget( QString, QString ) ) );
+    QDBusConnection sysBus = QDBusConnection::systemBus();
+    if ( !sysBus.isConnected() ) {
+	qFatal( "Cannot connect to the D-Bus system bus" );
+	exit( 1 );
+    }
+    sysBus.connect( QString(), QString(), "org.opensuse.OCIHelper", "displayProblemAndSolutions", this, SLOT( problemSolutionWidget( QString, QString ) ) );
+   
+    setLayout ( m_mainLayout );
+}
+
+void ConflictResolutionScreen::problemSolutionWidget( QString probDescription, QString solutions )
```

Comment:
Have you tried using a QStringList class for solutions instead of a QString with '$' separators ?
Instead of 
for ( QString solution : solutions.split( "$", QString::SkipEmptyParts) ) {
You might use later:
for ( QString solution : solutions ) {